packages/flutter/lib/src/widgets/raw_menu_anchor.dart

```diff
@@ -173,19 +242,94 @@ class RawMenuAnchor extends StatefulWidget {
+  ///
+  /// This callback is not called when an already open menu is repositioned.
   final VoidCallback? onOpen;
 
-  /// A callback that is invoked when the menu is closed.
+  /// Called when the menu overlay is hidden.
+  ///
+  /// This callback is triggered when the menu overlay is removed from the
+  /// widget tree, typically after any closing animations have completed.
+  ///
+  /// If a callback is provided for [onCloseRequested], [onClose] will not be
+  /// called until after `hideOverlay` is called.
   final VoidCallback? onClose;
 
+  /// Called when a request is made to open the menu.
+  ///
+  /// This callback is used by themed menu widgets to intercept open requests,
+  /// typically with the intent of adding a delay or an animation.
+  ///
+  /// After an open request is intercepted, the handler is responsible for
+  /// eventually calling `showOverlay` to add the menu overlay to the widget
+  /// tree and start the opening animation.
+  ///
+  /// The `showOverlay` call displays the overlay and generally precedes the
+  /// opening animation, since the animation plays on the overlay itself. After
+  /// this call, [MenuController.isOpen] becomes true, and the menus are
+  /// expected to be interactive.
+  ///
+  /// The `position` argument is the position passed to [MenuController.open].
+  /// Handlers should provide this argument to `showOverlay` so that the
+  /// `overlayBuilder` can use it to position the menu. When a menu is
+  /// repositioned, [onOpenRequested] may be called with a new `position` value
+  /// while the menu is still open. In this case, opening delays or animations
+  /// should be skipped, and `showOverlay` should be called with the new
+  /// `position` value.
+  ///
+  /// A typical [onOpenRequested] consists of the following steps:
+  ///
+  ///  1. Optional delay.
+  ///  2. Call `showOverlay` (whose call chain eventually invokes [onOpen]).
+  ///  3. Optionally starts the opening animation.
+  ///
+  /// While [onOpenRequested] will always be called when [MenuController.open]
+  /// is called, [onOpen] will only be called when a closed menu is shown, or
+  /// when a menu is repositioned.
+  ///
+  /// Defaults to a callback that immediately shows the menu.
+  final RawMenuAnchorOpenRequestedCallback onOpenRequested;
+
+  /// Called when a request is made to close the menu.
+  ///
+  /// This callback is used by themed menu widgets to intercept close requests,
+  /// typically with the intent of adding a delay or an animation before the
+  /// menu is hidden.
+  ///
+  /// After a close request is intercepted, handlers are responsible for
+  /// eventually calling `hideOverlay` to remove the menu from the widget tree.
+  /// This should typically be after closing animations have completed.
+  ///
+  /// [MenuController.isOpen] will be set to false after `hideOverlay` is
+  /// called, meaning [MenuController.isOpen] will typically be true while
+  /// closing animations are running.
```

Comment:
Similar to above, I rewrote this part:
```suggestion
  /// After a close request is intercepted, handlers are responsible for
  /// eventually calling `hideOverlay`. If needed, the call can be made after a
  /// delay, or not at all. The `hideOverlay` call sets [MenuController.isOpen]
  /// to false and ensures that the menu overlay is hidden. If the
  /// `RawMenuAnchor` is used in a themed menu that plays a closing animation,
  /// `hideOverlay` should be called after the closing animation has ended,
  /// since the animation plays on the overlay itself. This means that
  /// [MenuController.isOpen] will true while closing animations are running.
```
Mostly I described in more details what `hideOverlay` does and grouped them into one sentence.